### PR TITLE
CIDR Support in vertx-pg-client

### PR DESCRIPTION
**Pull request by AyanKoche**
Motivation: CIDR Data type support in vertx-pg-client

Explain here the context, and why you're making that change, what is the problem you're trying to solve.
There are several data types supported by vertx-pg-client but CIDR was missing implementation. As the project I am working on requires that support, contributing it is the better way. 
Changes:
- Have added a CIDR class
- Have added 3 methods : binaryEncodeCidr, binaryDecodeCidr, textDecodeCidr
- Have added tests for CIDR class
- Have made relevant changes to the docs as well

Conformance:

You should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

File: vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java [hunk 1]
```diff
@@ -1713,4 +1720,92 @@ private static <T> void textEncodeArray(T[] values, DataType type, ByteBuf buff)
...
+    int size = buff.getByte(index+3);
+    byte[] data = new byte[size];
+    buff.getBytes(index+4,data);
+    InetAddress address;
+
+    switch (family){
+      case 2:
+        // IPV4
+        try {
+          address = Inet4Address.getByAddress(data);
```
Comment (vietj) [inline vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java:1737]: Actually `InetAddress.getByAddress`

File: vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java [hunk 2]
```diff
@@ -1713,4 +1720,92 @@ private static <T> void textEncodeArray(T[] values, DataType type, ByteBuf buff)
...
+        }catch (UnknownHostException e){
+          throw new DecoderException(e);
+        }
+        break;
+      case 3:
+        // IPV6
+        try {
+          address = Inet6Address.getByAddress(data);
```
Comment (vietj) [inline vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java:1745]: Actually `InetAddress.getByAddress`

File: vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java [hunk 3]
```diff
@@ -1713,4 +1720,92 @@ private static <T> void textEncodeArray(T[] values, DataType type, ByteBuf buff)
...
+    byte netmask = buff.getByte(index+1);
+    Integer val;
+    int size = buff.getByte(index+3);
+    byte[] data = new byte[size];
+    buff.getBytes(index+4,data);
+    InetAddress address;
+
+    switch (family){
+      case 2:
+        // IPV4
```
Comment (vietj) [inline vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java:1735]: IPV4 and IPV6 seems to be the same code and could be

if (family == 2 || family == 3) instead

**Review by vietj — CHANGES_REQUESTED** [review]
a few concerns to address

File: vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java (hunk 1 above)
Comment (AyanKoche) [reply vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java]: Addressed.

File: vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java (hunk 2 above)
Comment (AyanKoche) [reply vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java]: Addressed.

File: vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java (hunk 3 above)
Comment (AyanKoche) [reply vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java]: Addressed.

**Comment by AyanKoche** [comment]
Addressed the previous comments. Please re-review it. @vietj

**Review by vietj — CHANGES_REQUESTED** [review]

**Comment by vietj** [comment]
can you squash your commits in a single one ?

**Comment by AyanKoche** [comment]
Sure ,doing it

**Comment by AyanKoche** [comment]
@vietj Hi sorry for delayed response,  I have made the changes and squashed the commits into single commit. Kindly have a look at it

**Comment by AyanKoche** [comment]
Hi @vietj , any review comments on the PR?

**Review by vietj — CHANGES_REQUESTED** [review]
I cannot see any tests with a real database and the CIDR type, we need to have at least a test that read / write CIDR objects to a test container postgresql instance. Look at how other specific types are tested.

**Comment by vietj** [comment]
great feature thanks @AyanKoche

**Pull request merged**